src/client/telemetry/index.ts

```diff
@@ -42,19 +42,13 @@ function isTelemetrySupported(): boolean {
     }
 }
 
-type ErrorClassifier = (error: Error) => string | undefined;
-const errorClassifiers: ErrorClassifier[] = [];
-export function registerErrorClassifier(classifier: ErrorClassifier) {
-    errorClassifiers.push(classifier);
+type ErrorTelemetryUpdater = (props: Record<string, any>, error: Error) => void;
+const errorTelemetryUpdaters: ErrorTelemetryUpdater[] = [];
+export function registerErrorTelemetryUpdater(classifier: ErrorTelemetryUpdater) {
+    errorTelemetryUpdaters.push(classifier);
 }
-function getErrorClassification(error: Error): string {
-    for (const classifier of errorClassifiers) {
-        const classification = classifier(error);
-        if (classification && classification !== 'unknown') {
-            return classification;
-        }
-    }
-    return 'unknown';
+function populateTelemetryWithErrorInfo(props: Record<string, any>, error: Error) {
+    errorTelemetryUpdaters.forEach(item => item(props, error));
```

Comment:
Generic way to populate telemetry with details of the 
* Error (type)
* Python stack trace (without PII, hash of file & folder)